libqtile/lazy.py

```diff
@@ -69,20 +69,38 @@ def kwargs(self) -> Dict:
         """The kwargs to the given call"""
         return self._kwargs
 
-    def when(self, layout=None, when_floating=True):
-        self._layout = layout
+    def when(self, layout: Optional[Union[Iterable[str], str]] = None,
+             when_floating: bool = True) -> 'LazyCall':
+        """Enable call only for given layout(s) and floating state
+
+        Parameters
+        ----------
+        layout : str, Iterable[str], or None
+            Restrict call to given layout names. If None, enable call
+            for all layouts.
+            Can be a single name, a string of comma-separated names, or any
```

Comment:
As long as you are allowing iterables of names, I'd say to drop the parsing of comma separated names.

Reply:
Is there a drawback supporting both?
The syntax with comma separated names is lighter and can therefore be more attractive to some. Fair enough, people could call ``layout="max,columns".split(",")`` themselves but it doesn't cost much to have this on the library side. Or does it?

Reply:
`layout="max,columns"` is only four characters lighter than `layout=("max", "columns")` (five if you count the space) so it doesn't seem like a useful shortcut. Let's use real Python structures and keep a clean interface.

Reply:
Fair enough, fixed by a1d2fce.